runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -236,6 +281,126 @@ function M.on_publish_diagnostics(_, result, ctx, config)
   vim.diagnostic.set(namespace, bufnr, diagnostic_lsp_to_vim(diagnostics, bufnr, client_id))
 end
 
+---@private
+local function resolve_bufnr(bufnr)
+  return bufnr > 0 and bufnr or api.nvim_get_current_buf()
+end
+
+--- Refresh diagnostics for a buffer
+---
+---@param opts (nil|table) Optional arguments
+---  - bufnr (integer, default: 0): Buffer whose hints to refresh
+---  - only_visible (boolean, default: false): Whether to only refresh diagnostics for the visible regions of the buffer
+---
+---@private
+function M.refresh(opts)
```

Comment:
```suggestion
function M._refresh(opts)
```

Or `local function`, does it need to be exposed?